gulpfile.js

```diff
@@ -148,11 +151,46 @@ function prebidSource(webpackCfg) {
   const analyticsSources = helpers.getAnalyticsSources();
   const moduleSources = helpers.getModulePaths(externalModules);
 
-  return gulp.src([].concat(moduleSources, analyticsSources, 'src/prebid.js'))
+  return gulp.src([].concat(moduleSources, analyticsSources, helpers.getPrecompiledPath('src/prebid.js')))
     .pipe(helpers.nameModules(externalModules))
     .pipe(webpackStream(webpackCfg, webpack));
 }
 
+const PRECOMP_TASKS = new Map();
+function babelPrecomp({distUrlBase = null, disableFeatures = null, dev = false} = {}) {
+  if (dev && distUrlBase == null) {
+    distUrlBase = argv.distUrlBase || '/build/dev/'
+  }
+  const key = `${distUrlBase}::${disableFeatures}`;
+  if (!PRECOMP_TASKS.has(key)) {
+    const babelConfig = require('./babelConfig.js')({
+      disableFeatures: disableFeatures ?? helpers.getDisabledFeatures(),
+      prebidDistUrlBase: distUrlBase ?? argv.distUrlBase
+    });
+    const precompile = function() {
+      return gulp.src(SOURCE_PAT, {base: '.', since: gulp.lastRun(precompile)})
```

Comment:
Why is  this `since` necessary? Is this a caching mechanism of gulp or does it set a timestamp on the generated files? Or both?

Reply:
It only matters for the watch (development) tasks, to recompile only the files that change. Caching is within one gulp execution so "normal" builds are not affected by this.

It's also the reason for reusing tasks (your other question above), since the cache key is the task object.